lib/koa-authenticator.js

```diff
@@ -13,11 +14,15 @@ module.exports = function(escherConfig = {}, logger) {
         yield next;
       }
     } catch (ex) {
-      if (logger && logger.error) {
-        logger.error('authentication_request_error', ex.message, ex);
+      if (ex instanceof AuthenticationError) {
+        if (logger && logger.error) {
+          logger.error('authentication_request_error', ex.message, ex);
+        }
+
+        this.throw(401, ex.message);
       }
 
-      this.throw(401, ex.message);
+      this.throw(ex);
```

Comment:
Why do we convert all errors to Koa errors? If there is something wrong in the request handling then it will be an exposed koa error instead of a normal one. What do you think @gabor-balla-emarsys ?